src/cache_cf.cc

```diff
@@ -920,6 +920,21 @@ configDoConfigure(void)
         }
     }
 
+    // check for duplicated entries, e.g. we won't have duplicated items like this:
+    // http_port 3128
+    // http_port 3128 ssl-bump ...
+    for (AnyP::PortCfgPointer s = HttpPortList; s != NULL; s = s->next) {
+        for (AnyP::PortCfgPointer s2 = s->next; s2 != NULL; s2 = s2->next) {
+            if (s->name && s2->name && strcmp(s->name, s2->name) != 0)
+                continue;
+            if (s->transport.protocol == s2->transport.protocol) {
+                debugs(3, DBG_CRITICAL, "ERROR: multiple definitions for http_port: " << s->name << ' ' << s->transport.protocol);
+                self_destruct();
```

Comment:
Please throw (the error) on configuration errors instead of calling the deprecated self_destruct(). Bad reconfigurations should be rejected instead of killing Squid, and using exceptions helps achieve that long-term goal.

For error handling examples, see `git grep throw src/cache_cf.cc` but throw a TextException object instead of using the deprecated TexcHere() macro. You can find many examples of thrown TextExceptions in the code as well.